components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -49,6 +56,48 @@ using ContentGroup = std::pair<std::string, bool>;
 constexpr char kAllContentGroup[] = "all";
 constexpr float kSampleContentGroupAllRatio = 0.2f;
 
+struct FeedGenerationInfo {
+  std::string locale;
+  FeedItems feed_items;
+  Publishers publishers;
+  std::vector<std::string> channels;
+  Signals signals;
+  std::vector<std::string> suggested_publisher_ids;
+  TopicsResult topics;
```

Comment:
Frankly, I missed where is the original `base::Value` from https://brave-today-cdn.brave.com/brave-today/feed.en_USjson that we got from DecodeService.

It seems it's already been processed on UI thread and been put to `this` as `topics_`, etc.
Because the list is huge (3500+ items) even just per-item copying it could result UI freeze..

Any options to completely process the huge data on non UI thread?

Reply:
Even on `master` with disabled `FeedV2Builder::GenerateFeed` the UI task takes 100ms on win-Static build ;(
It means that this PR only a part of a story. Also https://github.com/brave/brave-core/pull/21833#issuecomment-1921029106 should be taken into account.
<img width="579" alt="image" src="https://github.com/brave/brave-core/assets/45488748/2468de69-d347-43cd-bbd4-ccb02e074ac6">

<img width="1208" alt="image" src="https://github.com/brave/brave-core/assets/45488748/5a10898d-080e-4b22-ad55-42d79bf939c8">

Reply:
Agreed that we should push everything else into a background thread - that said, currently the `PublisherParsing` function you linked to seems to take 2ms, so I don't think it's urgent.

After a bit of investigation, I think the other 100ms are coming from the `FeedFetcher`. I'm going to try and push that into a background thread too.

Reply:
Thanks for fixing `FeedFetcher`.
If the parsed data (std::vectors) was small enough we would stop here.  Unfortunately, it seems even manipulating the parsed data on UI thread could be a problem: https://github.com/brave/brave-core/pull/21833#discussion_r1477747653